Runtime/Services/BacktraceDatabaseContext.cs

```diff
@@ -147,6 +145,21 @@ public BacktraceDatabaseRecord Add(BacktraceData backtraceData)
             return Add(record);
         }
 
+        private BacktraceDatabaseRecord GetRecordByHash(string hash)
+        {
+            for (int batchIndex = 0; batchIndex < BatchRetry.Count; batchIndex++)
```

Comment:
Why does GetRecordByHash need to iterate through all the records? Maybe this piece is not called often or not time-critical? Otherwise it seems there is room to optimize here (why can't the hash take us directly to the record we want without having to iterate through all records?)

Reply:
Database objects stores record in batches - each of them represents how many retries for each record we tried to do. Long story short:
every report in batch with index 0 represents all reports that are ready to send now,
every report in batch with index n != 0 represents all reports that we tried to send to Backtrace but because of any reason we failed (for example networking problem)

Optimization with accessing reports by using hash makes sense - however, this will require to change data structure and change other database methods. We won't get too much with this change because of the size of report database and deduplication rules .What do you think?

Reply:
Ah okay that makes sense, I appreciate that explaination!